solr/core/src/java/org/apache/solr/cloud/ShardLeaderElectionContextBase.java

```diff
@@ -179,7 +183,15 @@ void runLeaderProcess(boolean weAreReplacement, int pauseBeforeStartMs)
           ZkStateReader.STATE_PROP, Replica.State.ACTIVE.toString());
       assert zkController != null;
       assert zkController.getOverseer() != null;
-      zkController.getOverseer().offerStateUpdate(Utils.toJSON(m));
+      DocCollection coll = zkStateReader.getCollection(this.collection);
+      if (coll == null || coll.getStateFormat() < 2 || ZkController.sendToOverseer(coll, id)) {
```

Comment:
This isn't a function that was written in this PR, but asking in case you know - the "getStateFormat()" call appears to be an equality check (included in comment at bottom), is there a particular reason it's as generically named as it is? Mainly just asking because I can't tell if `getStateFormat()` indicates some subtlety beyond "the znodes match" even after digging in a bit

```
 public int getStateFormat() {
    return ZkStateReader.CLUSTER_STATE.equals(znode) ? 1 : 2;
 }
```

Reply:
(Just want to make sure if there is a particular subtle aspect there that I'm keeping it in mind while reviewing :) )